bftclient/src/matcher.h

```diff
@@ -66,6 +68,11 @@ class Matcher {
 
   void clearReplies() { matches_.clear(); }
 
+  void getPrimary(std::optional<ReplicaId>& primary) {
```

Comment:
I think the code would read better by returning the value rather than passing a mutable reference as an out parameter. It also wouldn't rely on the default value being set for that parameter.

You would just return `std::nullopt` for `config_.ignore_primary_` otherwise return `primary_`. No need to check the value of primary_.

Reply:
fixed